### DISCUSSION TITLE
Process driven workflow (using DNN features linked to a workflow engine).

### DISCUSSION THREAD

**Discussion by KenTombs** [dnnsoftware/Dnn.Platform#4646] [Ideas]
## Description of problem
The interest here is in being able to use customer business processes/workflows (BPMN2 notation), to drive features of DNN.  So for example, define a process diagrammatically that when running uses resources from DNN, like asking the user to complete a predefined form or populate an email, then have the results stored away for further work or reporting. 

## Description of solution
There are many approaches to how this capability might work, using logical statements and links between DNN resources and such. Much of this can be achieved already with the more mature extension forms packages; it is though ultimately programmatic bringing limitations in time and process origination for non-technical users. It also probably limits use to extensions that are 'wfm enabled'. 

## Description of alternatives considered

1. Pursue some of the tantalizing hints of what might already exist (???)

2. Implement an open source workflow engine as an extension (Activiti, Alfresco or similar).  DNN drives the workflow engine.

3. Implement an open source workflow engine separately and use an API to create the integration.  The workflow engine drives DNN.

4. Others??

## Screenshots
![image](https://user-images.githubusercontent.com/46649875/56811028-13228f00-6838-11e9-8647-4d4a61e6c213.png)

## Additional context
This function would make DNN more widely usable for application building purposes.

## Affected version
Latest versions onwards

## Affected browser
Na

**Comment by roman-yagodin** [comment]
A very long shot. I think this will require strict standardization in many areas, which could contradict the DNN goal (please correct me, if I'm wrong and there is no such thing) to make the platform friendly for wide spectre of web developers and technologies.

**Comment by valadas** [comment]
In the spirit of cleaning the backlog, is someone planning in implementing this in let's say the next year or so or is this just an idea and nobody would pick it up anytime soon ?

**Comment by KenTombs** [comment]
This is something planned for later this year, so current.

**Comment by valadas** [comment]
Thanks @KenTombs I will put this in the Dnn 10 milestone then

**Comment by stale** [comment]
We have detected this issue has not had any activity during the last 90  days. That could mean this issue is no longer relevant and/or nobody has  found the necessary time to address the issue. We are trying to keep the  list of open issues limited to those issues that are relevant to the  majority and to close the ones that have become 'stale' (inactive). If no  further activity is detected within the next 14 days, the issue will be  closed automatically.
If new comments are are posted and/or a solution (pull request) is  submitted for review that references this issue, the issue will not be  closed. Closed issues can be reopened at any time in the future. Please remember those participating in this open source project are volunteers  trying to help others and creating a better DNN Platform for all. Thank  you for your continued involvement and contributions!

**Comment by roman-yagodin** [comment]
Please do not close!

**Comment by david-poindexter** [comment]
@KenTombs @roman-yagodin is this still under development?  If so, can we get this broken into smaller, more manageable issues?  Thanks!

**Comment by roman-yagodin** [comment]
I think this at least deserves to be stored in the backlog, not just closing.

**Comment by valadas** [comment]
I think this should be moved to discussions until we have at least some breakdown of the requirements needed for such implementation. We have many great ideas in the issues backlog that are not really actionable issues until someone actually picks it up or a good discussion happens on the steps necessary to get there, hence David question about knowing if anyone is working on it or can break it down into more manageable and actionable issues. If it's just an idea and not more yet, great but it should be in the discussions.

**Comment by roman-yagodin** [comment]
Thinking on more manageable chunks, is there any human-friendly documentation for current DNN Workflow API? Could be nice first thing to have as a reference point for workflow-related discussions.

[This post](https://www.dnnsoftware.com/community-blog/cid/155185/implementing-workflow-in-a-custom-dnn-module) is the only thing I've found so far - but it looks more like the DnnHero promo article.

**Comment by valadas** [comment]
I am moving this into a discussion for now to clean the backlog, happy to bring it back when we hear some news about progress on this.
